SSO-Auth/Config/configPage.html

```diff
@@ -1,15 +1,272 @@
+                <div class="fieldDescription checkboxFieldDescription">
+                  Determines if the user logging in is allowed access to all
+                  folders.
+                </div>
+              </div>
+
+              <!-- Make this multiline, one per line and then split on newline -->
+
+              <div class="inputContainer">
+                <label
+                  class="inputLabel inputLabelUnfocused"
+                  for="EnabledFolders"
+                  >Enabled Folders:</label
+                >
+                <input is="emby-input" id="EnabledFolders" type="text" />
+                <div class="fieldDescription">
+                  JSON Array of Strings. If <strong>enableAllFolders</strong> is
+                  set to false, then this will be used to determine what folders
+                  the users who log in through this provider are allowed to use.
+                </div>
+              </div>
+
+              <div class="inputContainer">
+                <label class="inputLabel inputLabelUnfocused" for="Roles"
+                  >Roles:</label
+                >
+                <input is="emby-input" id="Roles" type="text" />
+                <div class="fieldDescription">
+                  This validates the OpenID response against the claim set in
+                  RoleClaim. If a user has any of these roles, then the user is
+                  authenticated. Leave blank to disable role checking. This
+                  currently only works for Keycloak (to my knowledge).
+                </div>
+              </div>
+
+              <div class="inputContainer">
+                <label class="inputLabel inputLabelUnfocused" for="AdminRoles"
+                  >Admin Roles:</label
+                >
+                <input is="emby-input" id="AdminRoles" type="text" />
+                <div class="fieldDescription">
+                  Like "Roles", but confers admin privelege. If unset will not
+                  grant admin priveleges.
+                </div>
+              </div>
+
+              <div class="checkboxContainer checkboxContainer-withDescription">
+                <label>
+                  <input
+                    id="EnableFolderRoles"
+                    name="EnableFolderRoles"
+                    type="checkbox"
+                  />
+                  <span>Enable role-based folders</span>
+                </label>
+                <div class="fieldDescription checkboxFieldDescription">
+                  Determines if user roles should be used to control library
+                  access.
+                </div>
+              </div>
+
+              <div class="inputContainer">
+                <label
+                  class="inputLabel inputLabelUnfocused"
+                  for="FolderRoleMapping"
+                  >Folder Role Mapping:</label
+                >
+                <input is="emby-input" id="FolderRoleMapping" type="text" />
```

Comment:
Should not take raw JSON and instead be more user-friendly.

Reply:
Resolved (In the sense that will be improved in later iteration)